packages/cli/cli-report.md

```diff
@@ -213,6 +213,8 @@ Options:
   --experimental-type-build
   --skip-build-dependencies
   --stats
+  --public-path <path>
+  --backend-url <url>
```

Comment:
Sorry, I just sat down and had a think about this again, and the question that popped up was, do we really need these?

I was suggesting some updates to documentation where we basically always recommend people to use `--public-path / --backend-url /`, unless they got deployments on a path. So then I'm wondering why we ask everyone to do that, why not just do it automatically?

Basically, could we take this new `app.baseUrl` rewrite that you just added to the frontend, and just always do it? Regardless of what the `app.baseUrl` is in config, we always update it to match `window.location.origin`? On top of that was also rewrite `backend.baseUrl`, but only if it has the same origin in config as `app.baseUrl`. As far as I can tell that'll pretty much always lead to the behavior that you want.

It's a pretty big 180 on the CLI config approach, but it's always a lot nicer if we can do the right thing out of the box straight away. Again, apologies that this might delay and cause even more iteration on this, but it's a tricky one 😅

Reply:
That definitely makes more sense to me and ends up being much less code! 

Using the `new URL(config, document.location.origin)` is all we'd need here (I think). It doesn't update the origin if it's different than the current document. Why would we want to check for same origin before updating `backend.baseUrl`? 

Also, I'm assuming we're getting rid of the CLI options with this new direction? Should I also add relative URL support to the config parser for `app.baseUrl` and `backend.baseUrl`? This would mean adding 'http://test-site.dev' to any `new URL` constructor.

Reply:
Yep!

If the backend is hosted separately from the frontend we'd break things if we rewrite the origin. Consider this config for example:

```yaml
app:
  baseUrl: https://backstage.acme.com

backend:
  baseUrl: https://backstage-backend.acme.com
```

If the origins match I'm thinking it should be safe to rewrite though.

And yep, we'd skip the CLI options. I don't think we *need* to add support for relative URLs for the frontend config, but I think we could.

Reply:
👍 , the other thing that came to mind is that this means the front end is going to need to maintain a separate config file, like 
```
app:
  baseUrl: '/test'
backend:
  baseUrl: '/'
```
is that expected? I know initially that was something we were trying to stay away from.

Reply:
I don't think so. There shouldn't be any need for the separate config as long as you want to keep the overall behavior that's present in the common config. I.e. if the common config has `app.baseUrl == backend.baseUrl`, then you can deploy the frontend on any origin without issues, as long as the backend is available on that same origin too.